### PR TITLE
Add user facing doc for management-components upgrade

### PR DESCRIPTION
**Pull request by d8660091**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+linkTitle: "Upgrade management components"
+weight: 21
+aliases:
+    /docs/tasks/cluster/cluster-upgrades/management-components-upgrade/
+date: 2024-02-21
+description: >
+  How to perform management components upgrade for management clusters
+---
+
+>**_NOTE:_** `upgrade management-components` sub-command is added in `eksctl anywhere` version `v0.19.0` for all providers. Management components upgrade can only be done through `eksctl-anywhere` cli, not through Kubernetes API.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:12]:
```suggestion
>**_NOTE:_** The `upgrade management-components` sub-command is added in `eksctl anywhere` version `v0.19.0` for all providers. Management components upgrades can only be done through the `eksctl anywhere` cli, not through the Kubernetes API.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### What are “management components”?
+
+Management components are the k8s resources in an EKSA management cluster that are responsible for managing the EKS-A clusters' lifecycle. Key components include, but are not limited to:
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:16]:
```suggestion
Management components are the Kubernetes resources in an EKS Anywhere management cluster that are responsible for managing the EKS Anywhere cluster's lifecycle. Key components include, but are not limited to:
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+* The `eksa-controller-manager` pod located in the `eksa-system` namespace.
+* Tinkerbell services located in a bare metal management cluster.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:19]:
```suggestion
* Tinkerbell services located in a Bare Metal management cluster.
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+* Custom Resource Definitions (CRDs) such as `clusters`, `eksareleases`, and `tinkerbellmachineconfigs`.
+
+### Why upgrade management-components separately?
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:22]:
```suggestion
### Why upgrade management components separately?
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+The “upgrade cluster” command, when executed against a management cluster, not only upgrades its management components but also initiates a comprehensive upgrade of the management cluster itself. This process often entails node replacement and significant operational tasks, which might be deemed risky for some production environments due to the potential of replacing all Kubernetes nodes.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:24]:
```suggestion
The `upgrade cluster` command, when executed against a management cluster, not only upgrades its management components but also initiates a comprehensive upgrade of the management cluster itself. This process often entails node replacement and significant operational tasks, which might be deemed risky for some production environments due to the potential of replacing all Kubernetes nodes. 
```

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+* Custom Resource Definitions (CRDs) such as `clusters`, `eksareleases`, and `tinkerbellmachineconfigs`.
+
+### Why upgrade management-components separately?
+
+The “upgrade cluster” command, when executed against a management cluster, not only upgrades its management components but also initiates a comprehensive upgrade of the management cluster itself. This process often entails node replacement and significant operational tasks, which might be deemed risky for some production environments due to the potential of replacing all Kubernetes nodes. 
+
+However, to leverage enhancements of newer EKSA version, such as bug fixes in the new management components without the extensive risk, it's advisable to initially upgrade only the management components. This step ensures minimal disruption and risk. Subsequently, you can proceed to upgrade the workload cluster at a more convenient time, once you're confident in the stability and readiness of your environment. This strategy provides a more controlled and efficient method to update your cluster's management infrastructure, aligning upgrades with your readiness and risk tolerance levels.
```
Comment (chrisnegus) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:26]:
```suggestion
However, to leverage enhancements to newer EKS Anywhere versions, such as bug fixes in the new management components without the extensive risk, it's advisable to initially upgrade only the management components. This step ensures minimal disruption and risk. Subsequently, you can proceed to upgrade the workload cluster at a more convenient time, once you're confident in the stability and readiness of your environment. This strategy provides a more controlled and efficient method to update your cluster's management infrastructure, aligning upgrades with your readiness and risk tolerance levels.
```

**Review by chrisnegus — COMMENTED** [review]
This reads very well. Just a few wording suggestions.

**Review by csplinter — CHANGES_REQUESTED** [review]

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
+---
+title: "Upgrade management components"
+linkTitle: "Upgrade management components"
+weight: 21
+aliases:
+    /docs/tasks/cluster/cluster-upgrades/management-components-upgrade/
+date: 2024-02-21
+description: >
+  How to perform management components upgrade for management clusters
```
Comment (csplinter) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:9]: How to upgrade EKS Anywhere management components

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+>**_NOTE:_** The `upgrade management-components` sub-command is added in `EKS Anywhere` version `v0.19.0` for all providers. Management components upgrades can only be done through the `eksctl anywhere` cli, not through the Kubernetes API.
```
Comment (csplinter) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:12]: The `eksctl anywhere upgrade management-components` subcommand was added in EKS Anywhere version `v0.19.0` for all providers. Management component upgrades can only be done through the `eksctl` CLI, not through the Kubernetes API.

File: docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+Performing setup and validations
+✅ Docker provider validation
+✅ Control plane ready
+✅ Cluster CRDs ready
+Upgrading core components
+Installing new eksa components
+🎉 Management components upgraded!
+```
+
+At this point, a new `eksaVersion` custom resource will be available in your management cluster. You can subsequently run a workload cluster upgrade with the `eksctl anywhere upgrade cluster command`, or by updating `eksaVersion` field in your workload cluster's spec and applying it to your management cluster with Kubernetes API-compatible tooling such as kubectl, GitOps, or Terraform.
```
Comment (csplinter) [inline docs/content/en/docs/clustermgmt/cluster-upgrades/management-components-upgrade.md:74]: At this point, a new `eksaVersion` custom resource will be available in your management cluster, which means new cluster components that correspond to the `eksaVersion` are available for cluster upgrades.

**Review by csplinter — APPROVED** [review]

**Pull request merged**